src/Google.Cloud.Functions.Invoker.Tests/FunctionsStartupTest.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Functions.Framework;
+using Google.Cloud.Functions.Invoker;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+[assembly: FunctionsStartup(typeof(Google.Cloud.Functions.Invoker.Tests.FunctionsStartupTest.TestStartup1), Order = 3)]
+[assembly: FunctionsStartup(typeof(Google.Cloud.Functions.Invoker.Tests.FunctionsStartupTest.TestStartup2), Order = 2)]
+
+namespace Google.Cloud.Functions.Invoker.Tests
+{
+    /// <summary>
+    /// Tests for everything to do with FunctionsStartup.
+    /// </summary>
+    public partial class FunctionsStartupTest
+    {
+        // Startups used for assembly attribute detection.
+        public class TestStartup1 : FunctionsStartup { }
+        public class TestStartup2 : FunctionsStartup { }
+
+        // This is the only test that uses the assembly attribute detection.
+        // Everything else specifies the startups explicitly.
+        [Fact]
+        public void GetStartups()
+        {
+            var startups = HostingInternals.GetStartups(typeof(FunctionsStartupTest).Assembly);
+            var actualTypes = startups.Select(startup => startup.GetType()).ToArray();
+            // TestStartup2 comes before TestStartup1 due to the Order properties.
```

Comment:
This is what we can't ensure after we've added them to the Services Collection, getting them for executing them App Configuration might give them out in a different order.

Reply:
https://www.stevejgordon.co.uk/asp-net-core-dependency-injection-registering-multiple-implementations-interface includes:

> It’s worth making it clear that the implementations will be added in the order they are registered. They will be returned in that same order when injected into calling code. Depending on your requirements, this may be useful and important.

That's not official documentation, but it's at least encouraging.